### PR TITLE
wip(Leaner Tests): Initial effort

### PR DESCRIPTION
**Pull request by e0ipso**
Task: #138

* [x] Ready for review
* [ ] Ready for merge

This PR aims to use the built-in testing capabilities to a complete separate node.js based stack (as described in #138).

### CODE REVIEW

**Comment by e0ipso** [comment]
This is still not ready. However I'm leaving this here since I have to move to other tasks. Hopefully someone else will be able to grab this code and take it to completion.

File: composer.json [hunk 1]
```diff
@@ -42,6 +42,9 @@
             "drupal/remote_stream_wrapper": {
                 "Module installation fails when shipped as part of an install profile": "https://www.drupal.org/files/issues/2886691-2.patch"
             },
+            "drupal/material_admin": {
```
Comment (e0ipso) [inline composer.json:45]: We no longer need this.

File: tests/src/Functional/InstallationTest.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+   * @param \Drupal\Core\Url $url
+   *   URL to request.
+   * @param array $request_options
+   *   Request options to apply.
+   *
+   * @return \Psr\Http\Message\ResponseInterface
+   */
+  protected function request($method, Url $url, array $request_options) {
+    try {
+      $response = $this->httpClient->request($method, $url->toString(), $request_options);
```
Comment (dawehner) [inline tests/src/Functional/InstallationTest.php:89]: You can achieve the same using ```$request_options[\GuzzleHttp\RequestOptions::HTTP_ERRORS] = FALSE;```

File: .travis.yml [hunk 3]
```diff
@@ -32,32 +34,18 @@ branches:
...
-  # See https://stackoverflow.com/questions/41487659/nightwatch-selenium-socket-hang-up
-  - export DBUS_SESSION_BUS_ADDRESS=/dev/null
-  - sh -e /etc/init.d/xvfb start
-  - sleep 3 # give xvfb some time to start
   - composer run-script start-contenta $PROJECT_BASE_PATH --timeout=0 &
   - sleep 5 # give web server some time to start
 
 script:
-  - yarn run nightwatch
+  - php ${DEST_DIR}/web/core/scripts/run-tests.sh --php `which php` --verbose --color --url http://${WEB_HOST}:${WEB_PORT} ContentaInstaller
```
Comment (dawehner) [inline .travis.yml:48]: Is there a reason we are not using the phpunit test runner? This is so much better to use IMHO

Comment (e0ipso) [reply .travis.yml:48]: Nope. I just wanted to mimick both Thunder and the Drupal CI.

Comment (e0ipso) [reply .travis.yml:48]: I'll be happy using the PHPUnit test runner.

File: tests/src/Functional/InstallationTest.php (hunk 2 above)
Comment (e0ipso) [reply tests/src/Functional/InstallationTest.php:89]: Nice!

**Comment by dawehner** [comment]
I think we can mark it as a duplicate of #164 now

**Comment by dawehner** [comment]
Thank you for your work @e0ipso

**Pull request closed without merging**